File: OOPDraw/Form1.cs
using Nakov.TurtleGraphics;
using System;
using System.Collections.Generic;
using System.Windows.Forms;

namespace OOPDraw
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private List<Shape> shapes = new List<Shape>();

        private Shape mostRecent;

        private void Form1_MouseClick(object sender, MouseEventArgs e)
        {
            //Transform windows coordinates to Turtle coordinates
            float turtleX = e.X - Width / 2 + 8;
            float turtleY = Height / 2 - e.Y - 19;
            string selectedItem = (string)comboBox1.SelectedItem;
            if (selectedItem == "Draw Triangle") //We will add more options later
            {
                AddShape(new EquilateralTriangle(turtleX, turtleY, 50));
            }
            else if (selectedItem == "Draw Rectangle")
            {
                AddShape(new Rectangle(turtleX, turtleY, 100, 50));
            }
            else if (selectedItem == "Draw House")
            {
                AddShape(new House(turtleX, turtleY, 100, 50));
            }
            else if (selectedItem == "Move Shape")
            {
                ActiveShape().MoveTo(turtleX, turtleY);
            }
            else if (selectedItem == "Resize Shape")
            {
               ActiveShape().ResizeAbsolute(turtleX, turtleY);
            }
            DrawAll();
        }

        private void AddShape(Shape shape)
        {
            if (shapes.Count > 0) //i.e. this isn't the first shape
            {
                ActiveShape().Unselect();
            }
            shapes.Add(shape);
            activeShapeNumber = shapes.Count - 1; //i.e. the shape just added
            ActiveShape().Select();
        }

        public void DrawAll()
        {
            Turtle.Dispose();  //First clear all Turtle tracks to start afresh
            foreach (var shape in shapes)
            {
                shape.Draw();
            }
        }

        private int activeShapeNumber = 0;

        private Shape ActiveShape()
        {
            return shapes[activeShapeNumber]; //List elements can be accessed like an array
        }

        private void Next_Click(object sender, System.EventArgs e)
        {
            ActiveShape().Unselect();
            activeShapeNumber = activeShapeNumber + 1;
            if (activeShapeNumber >= shapes.Count) activeShapeNumber = 0;
            ActiveShape().Select();
            DrawAll();
        }

        private void Prev_Click(object sender, System.EventArgs e)
        {
            ActiveShape().Unselect();
            activeShapeNumber = activeShapeNumber - 1;
            if (activeShapeNumber < 0) activeShapeNumber = shapes.Count - 1;
            ActiveShape().Select();
            DrawAll();
        }
    }
}

File: OOPDraw VB/OOPDraw/Form1.vb
Imports Nakov.TurtleGraphics

Public Class Form1

    Private shapes As List(Of Shape) = New List(Of Shape)()

    Private Sub Form1_MouseClick(sender As Object, e As MouseEventArgs) Handles MyBase.MouseClick
        Dim turtleX As Single = CType(e.X - Width / 2 + 8, Single)
        Dim turtleY As Single = CType(Height / 2 - e.Y - 19, Single)
        Dim selectedItem As String = CType(ComboBox1.SelectedItem, String)
        If selectedItem = "Draw Triangle" Then 'We Then will add more options later
            AddShape(New EquliateralTriangle(turtleX, turtleY, 50))
        ElseIf selectedItem = "Draw Rectangle" Then
            AddShape(New Rectangle(turtleX, turtleY, 100, 50))
        ElseIf selectedItem = "Draw House" Then
            AddShape(New House(turtleX, turtleY, 100, 50))
        ElseIf selectedItem = "Move Shape" Then
            ActiveShape().MoveTo(turtleX, turtleY)
        ElseIf selectedItem = "Resize Shape" Then
            ActiveShape().ResizeAbsolute(turtleX, turtleY)
        End If
        DrawAll()
    End Sub

    'ActiveShape().Resize(turtleX, turtleY)

    Private Sub AddShape(shape As Shape)
        If shapes.Count > 0 Then
            ActiveShape().UnselectShape()
        End If
        shapes.Add(shape)
        activeShapeNumber = shapes.Count - 1  'i.e. the shape just added
        ActiveShape().SelectShape()
    End Sub

    Public Sub DrawAll()
        Turtle.Dispose()  'First clear all Turtle tracks To start afresh
        For Each shape In shapes
            shape.Draw()
        Next
    End Sub

    Private activeShapeNumber As Integer = 0

    Private Function ActiveShape() As Shape
        Return shapes(activeShapeNumber) 'List elements can be accessed Like an array
    End Function

    Private Sub NextShape_Click(sender As Object, e As EventArgs) Handles NextShape.Click
        ActiveShape().UnselectShape()
        activeShapeNumber = activeShapeNumber + 1
        If (activeShapeNumber >= shapes.Count) Then activeShapeNumber = 0
        ActiveShape().SelectShape()
        DrawAll()
    End Sub

    Private Sub PrevShape_Click(sender As Object, e As EventArgs) Handles PrevShape.Click
        ActiveShape().UnselectShape()
        activeShapeNumber = activeShapeNumber - 1
        If (activeShapeNumber < 0) Then activeShapeNumber = shapes.Count - 1
        ActiveShape().SelectShape()
        DrawAll()
    End Sub
End Class
